flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/utils/RexNodeExtractor.scala

```diff
@@ -345,8 +345,11 @@ class RexNodeToExpressionConverter(
         SqlTimestamp.fromEpochMillis(millisecond, nanoOfMillisecond).toLocalDateTime
 
       case TIMESTAMP_WITH_LOCAL_TIME_ZONE =>
-        val v = literal.getValueAs(classOf[java.lang.Long])
-        unixTimestampToLocalDateTime(v).atZone(timeZone.toZoneId).toInstant
+        val v = literal.getValueAs(classOf[TimestampString])
+        val millisecond = unixTimestampToLocalDateTime(v.getMillisSinceEpoch)
+          .atZone(timeZone.toZoneId).toInstant.toEpochMilli
+        val nanoOfMillisecond = SqlDateTimeUtils.getNanoOfMillisSinceEpoch(v.toString)
```

Comment:
get nano from instant?

Reply:
ditto, the `Instant` only contains millisecond part

Reply:
You can use `LocalDateTime#of(int year, Month month, int dayOfMonth, int hour, int minute, int second, int nanoOfSecond)` to construct LocalDateTime and then change its timezone?

Reply:
refactor by introducing TimestampStringUtils to convert between LocalDateTime and TimestampString